### PR TITLE
Added check for duplicate dataframe columns

### PR DESCRIPTION
**Pull request by philippjfr**
- [x] Addresses https://github.com/ioam/holoviews/issues/2476
- [x] Adds unit test

### CODE REVIEW

**Comment by philippjfr** [comment]
Ready to merge.

File: holoviews/core/data/pandas.py [hunk 1]
```diff
@@ -80,6 +81,16 @@ def init(cls, eltype, data, kdims, vdims):
...
                     data = data.copy()
                     data.insert(0, kdim, np.arange(len(data)))
+
+            col_counter = Counter(data.columns)
+            for d in kdims+vdims:
+                if isinstance(d, Dimension): d = d.name
+                count = col_counter[d]
+                if d in col_counter and count > 1:
+                    raise DataError('Dimensions may not reference duplicated DataFrame '
+                                    'columns, found %d %r columns. If you want to plot '
```
Comment (jbednar) [inline holoviews/core/data/pandas.py]: Maybe `columns (found %d %r columns).`

File: holoviews/core/data/pandas.py [hunk 2]
```diff
@@ -80,6 +81,16 @@ def init(cls, eltype, data, kdims, vdims):
...
+
+            col_counter = Counter(data.columns)
+            for d in kdims+vdims:
+                if isinstance(d, Dimension): d = d.name
+                count = col_counter[d]
+                if d in col_counter and count > 1:
+                    raise DataError('Dimensions may not reference duplicated DataFrame '
+                                    'columns (found %d %r columns). If you want to plot '
+                                    'a column against itself simply declare two dimensions '
+                                    'with the same name. '% (count, d), cls)
```
Comment (jlstevens) [inline holoviews/core/data/pandas.py]: I don't really see a need to use ``Counter`` for this. I think I would have done something like:

```python
columns = []
for d in kdims+vdims:
   d = d.name if isinstance(d, Dimension) else d
    if d in columns:
        raise DataError('Dimensions may not reference duplicated DataFrame '
                                     'column  %r. If you want to plot '
                                    'a column against itself simply declare two dimensions '
                                    'with the same name. '% d, cls)
   columns.append(d)
```

It is just as long as avoids an import.

Comment (philippjfr) [reply holoviews/core/data/pandas.py]: That logic is wrong, we're checking for referenced columns that are duplicated in the dataframe, not duplicate dimensions. Happy to rewrite it without ``Counter`` though.

**Comment by philippjfr** [comment]
Some issues with test data, seems some bad data was captured or merged at some point causing a bokeh elements test to fail, going to rebuild and merge this to keep other stuff moving. I've addressed all the comments and it's not a controversial PR.

**Pull request merged**